libs/core/algorithms/include/hpx/algorithms/traits/pointer_category.hpp

```diff
@@ -133,6 +137,34 @@ namespace hpx::traits {
                     iter_value_t<Dest>>,
                 general_pointer_tag>;
         };
+
+        // relocatabillity is true in almost all cases,
+        // an exception is std::mutex.
+        template <typename T>
+        struct is_relocatable
+        {
+            static constexpr bool value =
+                std::is_move_constructible_v<T> && std::is_destructible_v<T>;
+        };
+
+        template <typename T>
+        inline constexpr bool is_relocatable_v = is_relocatable<T>::value;
+
+        template <typename Source, typename Dest,
+            bool NonContiguous = !iterators_are_contiguous_v<Source, Dest>>
```

Comment:
Should we make this to not use double negation? That would mean something like:
```suggestion
            bool IsContiguous = iterators_are_contiguous_v<Source, Dest>>
```

Reply:
agreed, this happens in multiple places in the file, should I change the pre-exising stuff too?

Reply:
> agreed, this happens in multiple places in the file, should I change the pre-exising stuff too?

Sure, if it's not too much of a bother.

Reply:
Not at all, what do you think about the file location of `hpx::is_relocatable`?

Reply:
This could be a new file in this directory, I think.

Reply:
Thanks, I will work on that and get back to you